core/blockchain.go

```diff
@@ -2063,6 +2063,11 @@ func (bc *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
 		for _, tx := range newChain[i].Transactions() {
 			addedTxs = append(addedTxs, tx.Hash())
 		}
+		// Collect reborn logs due to chain reorg
```

Comment:
@MariusVanDerWijden Is it OK to move this logic back into this for loop? 

I saw you moved it out in your last PR, I'm not sure if this is accidental behavior or intentional. Please correct me if there is a reason for it and I can change it back. Just for cleaning up the code.

Reply:
Yes  the reason is memory consumption. If we do it here, the peak memory consumption is way higher since we need to keep the logs in memory at the same time as we keep the transactions in memory, moving it out allows the gc to collect the transaction ls before allocating the logs

Reply:
reverted.